lib/taste_tester/commands.rb

```diff
@@ -143,6 +143,7 @@ def self._run_on_hosts(method)
           loop do
             begin
               hostname = queue.pop(true)
+              next unless hostname
```

Comment:
why next?? end the thread. `break` the loop, exit the thread. There's no more hostnames in the queue.

Reply:
If there is no more hostname in the queue, `pop` throws. This means we un-queued something that was empty:
https://ruby-doc.org/core-2.5.0/Queue.html#method-i-pop
> If the queue is empty, the calling thread is suspended until data is pushed onto the queue. If non_block is true, the thread isn't suspended, and ThreadError is raised.

Reply:
ah, OK, that's what I was missing. In that case no need to check if hostname is nil.. but put a comment that you're passing `true` to `pop` so that ThreadError will be thrown when the queue is empty. Sorry I didn't understand.